static_frame/core/index_hierarchy.py

```diff
@@ -456,147 +674,186 @@ def _from_type_blocks(cls: tp.Type[IH],
         Returns:
             :obj:`IndexHierarchy`
         '''
+        if blocks.shape[1] == 1:
+            raise ErrorInitIndex('blocks must have at least two dimensions.')
 
-        depth = blocks.shape[1]
-
-        # minimum permitted depth is 2
-        if depth < 2:
-            raise ErrorInitIndex('cannot create an IndexHierarchy from only one level.')
-        if index_constructors is not None and len(index_constructors) != depth:
-            raise ErrorInitIndex('if providing index constructors, number of index constructors must equal depth of IndexHierarchy.')
-
-        depth_max = depth - 1
-        depth_pre_max = depth - 2
-
-        token = object()
-        observed_last = [token for _ in range(depth)]
-
-        tree: tp.Any = dict() # order assumed and necessary
-
-        idx_row_last = -1
-        for (idx_row, d), v in blocks.element_items():
-            if idx_row_last != idx_row:
-                # for each row, we re-set current to the outermost reference
-                current = tree
-                idx_row_last = idx_row
-
-            # shared implementation with from_labels ---------------------------
-            if d < depth_pre_max:
-                if v not in current:
-                    current[v] = dict() # order necessary
-                else: # can only fetch this node (and not create a new node) if this is the sequential predecessor
-                    if v != observed_last[d]:
-                        raise ErrorInitIndex(f'invalid tree-form for IndexHierarchy: {v} cannot follow {observed_last[d]} when {v} has already been defined.')
-                current = current[v]
-                observed_last[d] = v
-            elif d < depth_max: # premax means inner values are a list
-                if v not in current:
-                    current[v] = list()
-                else: # cannot just fetch this list if it is not the predecessor
-                    if v != observed_last[d]:
-                        raise ErrorInitIndex(f'invalid tree-form for IndexHierarchy: {v} cannot follow {observed_last[d]} when {v} has already been defined.')
-                current = current[v]
-                observed_last[d] = v
-            elif d == depth_max:
-                # if there are redundancies here they will be caught in index creation
-                current.append(v)
-            else:
-                # cannot happen with TypeBlocks
-                raise ErrorInitIndex('label exceeded expected depth', v) #pragma: no cover
-            # shared implementation with from_labels ---------------------------
+        def gen_columns() -> tp.Iterator[np.ndarray]:
+            for block in blocks:
+                yield block.values.ravel()
```

Comment:
I am confused why we need `ravel` here...

Reply:
I see. Is `column_1d_filter` better... or if you know it is `(n, 1)` you can just directly reshape. `ravel` is one of those "handle all cases" NumPy functions I try to avoid...